### ISSUE TITLE
Custom confirmation message

### ISSUE THREAD

**Issue by almet** [Kinto/formbuilder#158]

**Comment by sahildua2305** [comment]
This is a great idea! After a quick look, I can see it needs to be added in `formbuilder/components/RecordCreated.js`. Do you have any suggestions about where do we ask for this custom confirmation message input from the creator of the form?

**Comment by rlecellier** [comment]
I think that we need a form configuration page.
It will be available from the form save's confirmation screen (where we display both url and admin url) and from the admin page.

**Comment by sahildua2305** [comment]
Yes, that's what I have been thinking as well @rlecellier. Thanks for the reply.

**Comment by almet** [comment]
As usual for such choices, the best way to act is to ask to UX people. @natouille @QuentinRoy, where do you think we should ask for a custom confirmation message?

**Comment by DirtyF** [comment]
@almet a Settings view would be the place to customize your form (you also spoke about background, colors, and such, custom message could belong there)

**Comment by almet** [comment]
In terms of flow, I would actually display a "change your confirmation message" screen when clicking on "submit the form", rather than having the user click on a "settings" view (or maybe this screen could be the settings view, in which different actions would be displayed ?)

**Comment by Natouille** [comment]
@almet it's always great to have a confirmation that your form was sell sent and delivered. It should be a page after the creation of the form. It will invite the creator to fill the options that will help the UX for the end users (but of course, they wil be optionnal). It's a better way than a settings button that could be ignore by the form creators.

**Comment by DirtyF** [comment]
@almet I agree, settings are just like hamburger icons, a lazy way to hide useful informations.

**Comment by rlecellier** [comment]
Could we use the admin view as a _form save confirmation page_ ?
I would regroup form information and settings:
* urls
* settings
* results 

It could be two parts, information and results. 
Each editable informations have a "edit" button that display the edit form.
Each part have a toggle button. Informations are hide by default when coming from admin url, display by default when coming from form save.
Also, we can add a message like "please take some time to configure your form" when coming from save form.

**Comment by almet** [comment]
I think results should still be in a different page.

>  Each editable informations have a "edit" button that display the edit form.

I'm not sure what you're refering to. Can you elaborate? If you're thinking about e.g the confirmation message, I believe we could have it in an input instead, and update the information as soon as the user changes it (with a small check to tell it's been updated).

If possible, I would reduce the distinction between "when coming from screen A" and "when coming from screen B", and always display the same thing, so the user knows where to find the information.

> Also, we can add a message like "please take some time to configure your form" when coming from save form.
+1

**Comment by DirtyF** [comment]
Question: Could the confirmation message be treated like a field you add at the end of the form?

**Comment by sahildua2305** [comment]
I like your idea @DirtyF! That's what I was thinking as well.

To make things simpler, we should just add another entry just like `description` at the end (just about the `Submit your form` button). It's optional to change it just like `title` or `description` of the form. This will give a better overview of the flexibility to the end user.

**Comment by almet** [comment]
I personally prefer to have this feature implemented as part of the confirmation page. To me fields are, well, fields! And we shouldn't tie them to other contexts, especially because they would mean adding a lot of logic just for them: if it's not present then fallback to another message, if it's present there should be only one, etc.

**Comment by sahildua2305** [comment]
As a user, when I'm thinking of creating a form these are the things I will have in mind to edit besides the form fields / questions - 
- Form Title
- Form Description (providing more information about the form to the users)
- Form Confirmation Text (providing information about next steps to the users who submit the forms to keep them engaged or whatever the purpose I have)

According to me, if we're keeping one thing separate than the other two, we're breaking the flow in which user thinks. Or may be it's just me who's a weird user. I'm up for further discussion over this.

**Comment by DirtyF** [comment]
Interface should abstract complexity, not the other way around ;)

**Comment by almet** [comment]
> Interface should abstract complexity, not the other way around ;)

I surely agree with this :-) We probably think in different ways about what's the simplicity is.

Let me try to propose a slight change to the approach you're proposing: let's add a "confirmation message" at the end of the form, which appears only when one or more fields have been added. It should be editable the same way the title and the description are editable.

Would that work for you?

**Comment by sahildua2305** [comment]
@almet yes, that'd be good as well! 👍

**Comment by rlecellier** [comment]
I dont like this confirmation message in the builder interface. 
It's a different page that'll be dislay after the form being submited. i feel it confusing to have this "confirm message" field  about a secondary page in the same interface.
